codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -153,7 +152,15 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
         try {
             Http2Stream stream = connection.stream(streamId);
             if (stream == null) {
-                stream = connection.local().createStream(streamId, endOfStream);
+                try {
+                    stream = connection.local().createStream(streamId, endOfStream);
+                } catch (Throwable cause) {
+                    if (connection.remote().mayHaveCreatedStream(streamId)) {
+                        promise.tryFailure(new IllegalStateException("Stream no longer exists: " + streamId, cause));
```

Comment:
connection.remote().mayHaveCreatedStream(streamId) will also be false and throw cause...I think this issue is no bussniess about goway... what we expect is just return and done the stream rather than  throw any exception

BTW client send RST_STREAM but not send GO_AWAY